src/Repository/SettingRepository.php

```diff
@@ -121,4 +78,48 @@ public function findAllByChannelAndLocaleWithDefault(string $vendor, string $plu
 
         return $queryBuilder->getQuery()->getResult();
     }
+
+    private function addChannelConditionWithNull(QueryBuilder $queryBuilder, ChannelInterface $channel = null): void
+    {
+        $this->addColumnConditionWithNull($queryBuilder, 'channel', $channel);
+    }
+
+    private function addChannelConditionWithoutNull(QueryBuilder $queryBuilder, ChannelInterface $channel = null): void
+    {
+        $this->addColumnCondition($queryBuilder, 'channel', $channel);
+    }
+
+    private function addLocaleConditionWithNull(QueryBuilder $queryBuilder, ?string $localeCode = null): void
+    {
+        $this->addColumnConditionWithNull($queryBuilder, 'localeCode', $localeCode);
+    }
+
+    private function addLocaleConditionWithoutNull(QueryBuilder $queryBuilder, ?string $localeCode = null): void
+    {
+        $this->addColumnCondition($queryBuilder, 'localeCode', $localeCode);
+    }
+
+    /**
+     * @param mixed $value
+     */
+    private function addColumnCondition(QueryBuilder $queryBuilder, string $columnName, $value, string $additionalCondition = ''): void
+    {
+        if (null === $value) {
+            $queryBuilder->andWhere(sprintf('o.%s IS NULL', $columnName));
+
+            return;
+        }
+        $queryBuilder
+            ->andWhere(sprintf('o.%s = :%s', $columnName, $columnName) . $additionalCondition)
+            ->setParameter($columnName, $value)
+        ;
+    }
+
+    /**
+     * @param mixed $value
+     */
+    private function addColumnConditionWithNull(QueryBuilder $queryBuilder, string $columnName, $value): void
+    {
+        $this->addColumnCondition($queryBuilder, $columnName, $value, sprintf(' OR o.%s IS NULL', $columnName));
+    }
```

Comment:
A lot of method for nothing no ?

Reply:
yes, i wanted to avoid phpmd errors but the code is complicated